app/controllers/MovementsController.scala

```diff
@@ -56,10 +58,13 @@ class MovementsController @Inject()(
 )(implicit ec: ExecutionContext)
     extends BackendController(cc) {
 
+  private val atLeastOneMessageSent: Seq[MovementMessage] => Boolean =
+    _.exists(x => x.optStatus.contains(SubmissionSucceeded) || x.optStatus.isEmpty)
```

Comment:
Can the x.optStatus.isEmpty condition be removed?

Reply:
No, if there is a response from upstream we can shortcircuit out of the list traversal

Reply:
Shouldn't the x.optStatus.isEmpty be first in this condition? I'm not sure how it'll handle contains if it is empty.

Reply:
`None.contains` will return false